### PR TITLE
feat(vm): make default `trace` `ArrayList` capacity configurable using `comptime`

### LINKED ISSUE #30: feat: make default `trace` ArrayList capacity configurable using comptime
https://github.com/keep-starknet-strange/cairo-zig/blob/ef11dff02cab138de0a6405901e87ad5a12f93fe/src/vm/core.zig#L54

currently `4096` value is hard coded, make this configurable at compile time using `comptime`

### PR DESCRIPTION
**Pull request by tcoratger**
Should close #30.

### CODE REVIEW

File: src/vm/core.zig [hunk 1]
```diff
@@ -42,16 +42,15 @@ pub const CairoVM = struct {
     /// - `CairoVM`: The created VM.
     /// # Errors
     /// - If a memory allocation fails.
-    pub fn init(allocator: *const Allocator, config: Config) !CairoVM {
+    pub fn init(comptime capacity: usize, allocator: *const Allocator, config: Config) !CairoVM {
```
Comment (abdelhamidbakhta) [inline src/vm/core.zig:45]: please change to a more explicit name, it's more the `trace_entries_initial_capacity`
And I would put it after config in the orders of parameters.

**Review by AbdelStark — CHANGES_REQUESTED** [review]
left small comments

File: src/vm/core.zig (hunk 1 above)
Comment (tcoratger) [reply src/vm/core.zig]: @abdelhamidbakhta Sounds good, just pushed the change.

**Review by AbdelStark — APPROVED** [review]
lgtm

**Pull request merged**